test/OrchardCore.Commerce.Tests.UI/Tests/BasicTests/SecurityScanningTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Atata.WebDriverSetup;
+using Lombiq.Tests.UI.SecurityScanning;
+using Lombiq.Tests.UI.Services.GitHub;
+using Microsoft.CodeAnalysis.Sarif;
+using Newtonsoft.Json;
+using Shouldly;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace OrchardCore.Commerce.Tests.UI.Tests.BasicTests;
+
+public class SecurityScanningTests : UITestBase
+{
+    public SecurityScanningTests(ITestOutputHelper testOutputHelper)
+        : base(testOutputHelper)
+    {
+    }
+
+    [Fact]
+    public Task FullSecurityScanShouldPass() =>
+        // On GitHub when running from Windows, this test always fails with the following error:
+        // The `docker.exe pull softwaresecurityproject/zap-stable:2.14.0 --quiet` command failed with the output below.
+        // no matching manifest for windows/amd64 10.0.20348 in the manifest list entries
+        // This is because the Docker on the runner is in Windows mode but zap-stable only has Linux images.
+        GitHubHelper.IsGitHubEnvironment && OSInfo.IsWindows
+            ? Task.CompletedTask
+            : ExecuteTestAfterSetupAsync(
+                context => context.RunAndConfigureAndAssertFullSecurityScanForAutomationAsync(
+                    configuration => FalsePositive(
+                        configuration,
+                        10202,
+                        "Absence of Anti-CSRF Tokens: The ProductListPart-Filters intentionally uses a GET form. No XSS risk.",
```

Comment:
Open a ZAP issue about this and link it here.

Reply:
[I think this is intentional because they say never use GET forms](https://www.zaproxy.org/docs/alerts/10202/#:~:text=Do%20not%20use%20the%20GET%20method%20for%20any%20request%20that%20triggers%20a%20state%20change.) so there is nothing to report. However it's not applicable to our case because we use the GET form simply because it's the canonical way to update the query string parameters. I considered rewriting the [ProductListPart.Filters.cshtml](https://github.com/OrchardCMS/OrchardCore.Commerce/blob/issue/TDEAL-16/src/Modules/OrchardCore.Commerce/Views/ProductListPart.Filters.cshtml#L19) to use Javascript to do the exact same thing without a form, but then I remembered that I'm a human and not a slave of the machines. (refer to the outro of Killing in the Name as I was literally raging against the machine here)

Reply:
We use the form and the GET request not to change state but to search though, right?

Just to be sure, I recommend you include "please" when talking to ChatGPT :).

Reply:
You are right. Although I'm not sure what "state" means when HTML is a stateless protocol. The database maybe? This form doesn't change anything in the DB.

Also don't worry, I'm always nice to ChatGPT because it never tries to boss me around. :)

Reply:
"State" here always means server state, so like the DB, yes. OK then!